### PR TITLE
Add rust crates 'warp' and 'enclose'

### PR DESCRIPTION
**Pull request by piscisaureus**

### CODE REVIEW

**Comment by ry** [comment]
Cool - you got it to work ! Looks good.

 I think we should land this in the inspector branch (https://github.com/denoland/deno/pull/2696) rather than preemptively in master.

**Comment by piscisaureus** [comment]
> I think we should land this in the inspector branch (#2696) rather than preemptively in master.

I agree, although I'd suggest to keep the PRs separate for now. The people working on #2696 can merge this branch for the time being; when it is close to landing we can land this PR first.
(Reason is that it is relatively difficult to work with two repos in tandem; if it needs a rebase there's a good chance I can do it with this standalone branch, but not when intermingled with unrelated PRs).

File: build_extra/rust/rust.gni [hunk 1]
```diff
@@ -237,6 +238,17 @@ template("_rust_crate") {
       "--color=always",
     ]
 
+    if (defined(generated_source_dir)) {
+      args += [
+        # Some crates (e.g. 'typenum') generate source files and place them in
+        # the directory indicated by the 'OUT_DIR' environment variable, which
+        # is normally set by Cargo. This flag tells run.py to set 'OUT_DIR' to
+        # the path where the current crate can find its generated sources.
+        "--generated-source-dir=" +
```
Comment (piscisaureus) [inline build_extra/rust/rust.gni]: I'm wondering if it would be better to rename this to "--build-script-out-dir"

**Comment by ry** [comment]
merged in #2696

**Pull request closed without merging**